scripts/commerce.js

```diff
@@ -61,64 +61,173 @@ export const authPrivacyPolicyConsentSlot = {
+ * const allPlaceholders = await fetchPlaceholders();
+ *
+ * // Fetch placeholders with specific path
+ * const placeholders = await fetchPlaceholders('placeholders/auth.json');
+ *
+ * // Get all placeholders including newly fetched ones
+ * const updatedPlaceholders = await fetchPlaceholders();
  */
-export async function fetchPlaceholders(prefix = 'default') {
-  const overrides = getMetadata('placeholders') || getRootPath().replace(/\/$/, '/placeholders.json') || '';
-  const [fallback, override] = overrides.split('\n');
+export async function fetchPlaceholders(path) {
+  const rootPath = getRootPath();
+  const fallback = getMetadata('placeholders');
   window.placeholders = window.placeholders || {};
 
-  if (!window.placeholders[prefix]) {
-    window.placeholders[prefix] = new Promise((resolve) => {
-      const url = fallback || `${prefix === 'default' ? '' : prefix}/placeholders.json`;
-      Promise.all([fetch(url), override ? fetch(override) : Promise.resolve()])
-        // get json from sources
-        .then(async ([resp, oResp]) => {
-          if (resp.ok) {
-            if (oResp?.ok) {
-              return Promise.all([resp.json(), await oResp.json()]);
-            }
-            return Promise.all([resp.json(), {}]);
+  // Track pending requests to prevent duplicate fetches
+  window.placeholders._pending = window.placeholders._pending || {};
+
+  // Initialize merged results storage as a single merged object
+  window.placeholders._merged = window.placeholders._merged || {};
+
+  // If no path is provided, return the merged placeholders
+  if (!path) {
+    return Promise.resolve(window.placeholders._merged);
+  }
+
+  // Create cache key for this specific combination
+  const cacheKey = [path, fallback].filter(Boolean).join('|');
+
+  // Prevent empty cache keys
+  if (!cacheKey) {
+    return Promise.resolve({});
+  }
+
+  // Check if there's already a pending request for this combination
+  if (window.placeholders._pending[cacheKey]) {
+    return window.placeholders._pending[cacheKey];
+  }
+
+  // fetch placeholders
+  const fetchPromise = new Promise((resolve) => {
+    const promises = [];
+
+    // Helper function to get or create fetch promise for a single resource
+    const getOrCreateFetch = (url, resourceCacheKey) => {
+      // Check if already cached
+      if (window.placeholders[resourceCacheKey]) {
+        return Promise.resolve(window.placeholders[resourceCacheKey]);
+      }
+
+      // Check if already pending
+      if (window.placeholders._pending[resourceCacheKey]) {
+        return window.placeholders._pending[resourceCacheKey];
+      }
+
+      // Create new fetch promise
+      const resourceFetchPromise = fetch(url).then(async (response) => {
+        if (response.ok) {
+          const data = await response.json();
+          // Cache the response
+          window.placeholders[resourceCacheKey] = data;
+          return data;
+        }
+        return {};
```

Comment:
If an error occurs, do we want to at least warn? This is sort of silently failing.

Reply:
Fixed